src/Emulator/Peripherals/Peripherals/DMA/STM32DMA.cs

```diff
@@ -399,6 +429,8 @@ private static uint FromTransferType(TransferType transferType)
                 }
                 throw new InvalidOperationException("Should not reach here.");
             }
+            
+            public uint transferBacklog;
```

Comment:
I'm wondering if having a counter here is correct.

`DoPeripheralTransfer` uses transfer parameters set by `HandleConfigurationWrite`. This means that once a channel is enabled, all the queued transfers will use the same configuration (which I believe can be different than the one when the transfer was "scheduled").

Do you have an example where such a counter is required?

Reply:
This is a good point which I have not tested.

It is necessary in the case where the DMA runs out of space in a buffer while the uart is still receiving data. In its original state the DMA will miss some bytes. This happens because during the buffer swap procedure in the zephyr driver a UART interrupt register is reset which causes a read from the UART data register, dequeuing a value. Unless the amount of requested transfers is kept track of and processed when the DMA is re-enabled and before the UART is reconfigured, then some data is lost.

I am not  100% sure what the real DMA hardware does in this case, but in the figure I mention in the PR description it shows a request being generated which I assumed would be queued/buffered somehow. 

I will investigate

Reply:
@mateusz-holenko

Without the counter, wouldn't the dma just do the same thing but also lose the data that my problem fixes? 

If data is being streamed and the DMA is reconfigured halfway through, then the second half will have the new settings.